### PR TITLE
Add rnnoise_get_frame_size()

### PR DESCRIPTION
**Pull request by lu-zero**
It provides a mean to know how many samples are processed at time.

### CODE REVIEW

File: include/rnnoise.h [hunk 1]
```diff
@@ -48,18 +48,60 @@
...
 
+/**
+ * Free a DenoiseState produced by rnnoise_create.
+ *
+ * The optional custom model must be freed by rnnoise_model_free() after.
+ */
 RNNOISE_EXPORT void rnnoise_destroy(DenoiseState *st);
 
+/**
+ * Denoise a frame of sample
```
Comment (tmatth) [inline include/rnnoise.h:87]: nit: samples

File: include/rnnoise.h [hunk 2]
```diff
@@ -48,18 +48,60 @@
...
+ * Load a model from a file
+ *
+ * It must be deallocated with rnnoise_model_free()
+ */
 RNNOISE_EXPORT RNNModel *rnnoise_model_from_file(FILE *f);
 
+/**
+ * Free a custom model
+ *
+ * It must be called after all the DenoiseState referring to it are freed.
```
Comment (tmatth) [inline include/rnnoise.h:103]: nit `DenoiseStates`

File: include/rnnoise.h [hunk 3]
```diff
@@ -48,18 +48,60 @@
 typedef struct DenoiseState DenoiseState;
 typedef struct RNNModel RNNModel;
 
+/**
+ * Return the size of DenoiseState
+ */
 RNNOISE_EXPORT int rnnoise_get_size();
 
+/**
+ * Return the number of samples processed by rnnoise_process_frame at time
```
Comment (tmatth) [inline include/rnnoise.h:57]: nit: `at a time`

**Comment by petterreinholdtsen** [comment]
Can you tell more about why you need this method?  Perhaps more information can convince the lead developer to include it?

**Comment by tmatth** [comment]
> Can you tell more about why you need this method? Perhaps more information can convince the lead developer to include it?

I'm guessing it's to avoid clients have to know internals about the library, see e.g.,:
http://git.videolan.org/?p=vlc.git;a=blob;f=modules/audio_filter/rnnoise.c;h=34229f0012487ffe00f932238b8d990e861e3b22;hb=HEAD#l77

**Comment by lu-zero** [comment]
Exactly what @tmatth said :)

**Comment by lu-zero** [comment]
Documentation errors addressed.

**Review by tmatth — APPROVED** [review]

**Comment by petterreinholdtsen** [comment]
Right, so at least VLC need the value, and have to hardcode it because it is not available from the library.  Sound convincing to me.

**Pull request merged**